apps/fishing-map/features/datasets/datasets.utils.ts

```diff
@@ -393,6 +395,24 @@ export const getEventsDatasetsInDataview = (dataview: UrlDataviewInstance) => {
   })
 }
 
+export type DatasetSchemaGeneratorProps = {
+  data: CSV
+  meta: ParseMeta
+}
+export const getDatasetSchema = ({ data, meta }: DatasetSchemaGeneratorProps) => {
```

Comment:
As the generic dataset.utils file is big enough, so what about moving all the custom upload functions to its own file within the `upload` folder?

Also, I think the signature of this should be something like `getDatasetSchemaFromCSV`